examples/kind-multinode-kcp/templates/k8s/konnectivity-agent-ds.yaml

```diff
@@ -0,0 +1,95 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: konnectivity-agent
+  namespace: kube-system
+  labels:
+    kubernetes.io/cluster-service: "true"
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    k8s-app: konnectivity-agent
+  namespace: kube-system
+  name: konnectivity-agent
+spec:
+  selector:
+    matchLabels:
+      k8s-app: konnectivity-agent
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        k8s-app: konnectivity-agent
+    spec:
+      priorityClassName: system-cluster-critical
+      tolerations:
+        - key: "CriticalAddonsOnly"
+          operator: "Exists"
+        - operator: "Exists"
+          effect: "NoExecute"
+      nodeSelector:
+        kubernetes.io/os: linux
```

Comment:
Could add `node-role.kubernetes.io/worker: ''` to prevent agents on control plane nodes?

Reply:
Done!

Reply:
Actually, seems like setting this label on the node in the `kind.config` breaks the cluster for some reason. `kind` doesn't label the worker nodes as such by default for some reason, and `kubelet` doesn't start if I provide a custom role label. Leaving this out for now.

Reply:
Did some more testing and opened an upstream issue: https://github.com/kubernetes-sigs/kind/issues/3657

Reply:
kubelet doesn't permit adding arbitrary lables in the k8s.io / kubernetes.io namespace because those are reserved for Kubernetes and need to be approved by the API review team.

You can instead use taints and tolerations to schedule only to worker nodes. 

Kind removes the control plane taint only if there is a single control plane node and no other nodes (so a one node cluster), otherwise the control plane node(s) are tainted, see:
https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/#control-plane-node-isolation

Reply:
@jkh52 Is it ok if I leave this as-is? I haven't noticed proxy agents getting spun up on the control plane anyway.

Reply:
No objections. Are taints + tolerations the reason that you don't see agents on control plane? Did you try a one-node cluster as well?

Reply:
On a one-node cluster, that node is the worker and the control plane host, so scheduling user workloads and control plane workloads would be expected, though I'm not sure it makes sense to even try deploying this application to a one node cluster ..?

On a multi-node cluster there's the well-known kubeadm taint on control plane nodes, unless you tolerate _all_ taints or that [taint](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/) specifically your pod won't schedule there. It's pretty typical that a cluster with registered control plane nodes would have a similar taint if not the same taint.